### PR TITLE
[SDK] Fix `project.set_artifact` not validating artifact exists

### PR DESCRIPTION
**Pull request by quaark**
Fix for https://jira.iguazeng.com/browse/ML-3168

### CODE REVIEW

File: mlrun/projects/project.py [hunk 1]
```diff
@@ -1164,17 +1170,24 @@ def _get_hexsha(self):
...
         return None
 
-    def get_item_absolute_path(self, url: str) -> typing.Tuple[str, bool]:
-        in_context = False
+    def get_item_absolute_path(
+        self,
+        url: str,
+        # check_absolute_path is a temporary parameter to allow for backwards compatibility
+        check_absolute_path: bool = False,
+    ) -> typing.Tuple[str, bool]:
```
Comment (Tankilevitch) [inline mlrun/projects/project.py:1178]: add docstring for the function, what is that boolean that being returned

File: mlrun/projects/project.py [hunk 2]
```diff
@@ -1164,17 +1170,24 @@ def _get_hexsha(self):
...
+            return url, False
+
+        # We don't want to change the url if the project has no cntext or if it is already absolute
+        in_context = self.spec.context and not url.startswith("/")
+        if in_context:
+            url = path.normpath(path.join(self.spec.get_code_path(), url))
+
+        if (not in_context or check_absolute_path) and not path.isfile(url):
+            raise OSError(f"{url} not found")
+
```
Comment (Tankilevitch) [inline mlrun/projects/project.py:1190]: use `mlrun.errors.MLRunNotFoundError`

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good, minor comments

File: mlrun/projects/project.py (hunk 2 above)
Comment (quaark) [reply mlrun/projects/project.py:1190]: cant... backwards compatibility.. Need `OSError` hear

File: mlrun/projects/project.py [hunk 3]
```diff
@@ -1164,17 +1170,30 @@ def _get_hexsha(self):
...
-            if self.spec.context and not url.startswith("/"):
-                in_context = True
-                url = path.normpath(path.join(self.spec.get_code_path(), url))
-                return url, in_context
-            if not path.isfile(url):
-                raise OSError(f"{url} not found")
+        if not url or "://" in url:
+            return url, False
+
+        # We don't want to change the url if the project has no cntext or if it is already absolute
```
Comment (alonmr) [inline mlrun/projects/project.py:1189]:
```suggestion
        # We don't want to change the url if the project has no context or if it is already absolute
```

File: tests/projects/test_project.py [hunk 4]
```diff
@@ -440,6 +440,34 @@ def test_set_function_with_relative_path(context):
...
+        ("handler.py", False, pytest.raises(OSError)),
+    ],
+)
+def test_set_artifact_validates_file_exists(
+    monkeypatch, artifact_path, file_exists, expectation
+):
+    artifact_key = "my-artifact"
+    project = mlrun.new_project("inline", context=str(assets_path()), save=False)
+
+    monkeypatch.setattr(
```
Comment (alonmr) [inline tests/projects/test_project.py:456]: why `monkeypatch`? I believe you can just give 1 existing file and 1 that doesn't

**Review by alonmr — COMMENTED** [review]
Nice job! minor typo and question

File: tests/projects/test_project.py (hunk 4 above)
Comment (quaark) [reply tests/projects/test_project.py:456]: Indeed, but there's no need to create real files for this unit test. I don't care about their content, I'm just checking the "file exists" functionality

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**